backend/src/models/criteria.ts

```diff
@@ -0,0 +1,18 @@
+import { Schema, model } from 'mongoose';
+
+const criteriaSchema = new Schema({
+  id: String,
```

Comment:
The API document had something slightly different for the schemas - here the schema defines the fields of the object as we know it, but the doc had these wrapped up in another class that just had id, user, and "the actual object" fields.
https://github.com/mcode/pathway-builder/wiki/Backend-API#collection-schemas

I'm assuming we always want to receive/return the regular object on webservice calls, but the wrapped object might make things easier for user/workspace grouping. Thoughts? (cc: @blangley28 )

Reply:
I updated the schemas/models so that all the models are now of the form

```
metadata: Object,
value: Object
```

where `value` is the actual pathway model (e.g. Criteria, Pathway, Workspace) and `metadata` is a catch all for things like `userid` for the user that owns it.  I didn't flesh the metadata out in any detail because right now we don't have any routes that provide that information.  I could, but we'd probably want to think about updating the API so we can actually drive that.

I'll update the wiki docs now.

Reply:
Would it be possible to promote the ID field to the top-level of each model as well? Maybe it's not as big a deal as I think but it seems like the ID is fundamental enough that it should exist at the top level

Reply:
I can do that.  Originally I had it added as a virtual field, but since those aren't saved to the database we can't query based on them.  I should be able to duplicate the field in the database so that we can search off it though.

Reply:
Added the id to the top level and updated the index & queries to be based on it.